### DISCUSSION TITLE
Disable SSR for the whole project?

### DISCUSSION THREAD

**Discussion by ghost** [vikejs/vike#1331] [Discussion] [answered]
Is i possible to disable SSR (server) and only enable SSG (client) for the entire project?

I removed `+onRenderHtml.tsx` from my project, in hope the it would just be ignored, but when I run build, it throws this error:

```bash
Error: [vike][Wrong Usage] No onRenderHtml() hook found, see https://vike.dev/onRenderHtml
    at getRenderHook (file:///home/arch/ent/teste-preact/node_modules/vike/dist/esm/node/runtime/renderPage/executeOnRenderHtmlHook.js:62:9)
    at executeOnRenderHtmlHook (file:///home/arch/ent/teste-preact/node_modules/vike/dist/esm/node/runtime/renderPage/executeOnRenderHtmlHook.js:12:36)
    at prerenderPage (file:///home/arch/ent/teste-preact/node_modules/vike/dist/esm/node/runtime/renderPage/renderPageAlreadyRouted.js:79:46)
    at file:///home/arch/ent/teste-preact/node_modules/vike/dist/esm/node/prerender/runPrerender.js:471:19
```

This is the basic tree:
```bash
.
├── index.html
├── package-lock.json
├── package.json
├── pages
│   ├── +config.h.ts
│   ├── assets
│   │   └── preact.svg
│   └── index
│       ├── +Page.tsx
│       └── style.css
├── public
│   └── vite.svg
├── renderer
│   ├── +onRenderClient.tsx
│   ├── PageShell.css
│   ├── PageShell.tsx
│   └── usePageContext.tsx
├── tsconfig.json
└── vite.config.ts
```

And this is my `+config.h.ts`:

```typescript
import type { Config } from 'vike';

export default {
  meta: {
    Page: {
      env: { server: false, client: true, }
    }
  }
} satisfies Config;
```

This was adapted from the result of `npm init preact`, with: `typescript`, `static generation` and `eslint`. And modified to use v1.0 of Vike.

----

I'm going to copy from `+onRenderHtml.tsx` from another project for now and ignore `dist/server/`.

I appreciate any help and the project. Sorry if I misunderstood any concept of the project.

**Comment by lourot** [comment, marked as answer]
> Is i possible to disable SSR (server) and only enable SSG (client) for the entire project?

Hi @RickStanley yes, this is achieved by:
1. keeping `+onRenderHtml.tsx` (you still want to render HTML *at build-time*)
2. enabling [pre-rendering](https://vike.dev/prerender-config)

See [the react-full-v1 example](https://github.com/vikejs/vike/tree/main/examples/react-full-v1). If you run `pnpm run build` in this example everything gets statically generated to `dist/client`. You could then just statically serve this folder.
